x/concentrated-liquidity/pool.go

```diff
@@ -164,10 +165,10 @@ func (k Keeper) CalculateSpotPrice(
 	}
 
 	if price.IsZero() {
-		return sdk.Dec{}, types.PriceBoundError{ProvidedPrice: price, MinSpotPrice: types.MinSpotPrice, MaxSpotPrice: types.MaxSpotPrice}
+		return sdk.Dec{}, types.PriceBoundError{ProvidedPrice: osmomath.BigDecFromSDKDec(price), MinSpotPrice: types.MinSpotPriceV2, MaxSpotPrice: types.MaxSpotPrice}
 	}
 	if price.GT(types.MaxSpotPrice) || price.LT(types.MinSpotPrice) {
```

Comment:
Is spot price still on 18 precision? How do we expect assets with spot price below the previous `MinSpotPrice` to be represented here?

Reply:
Yes, it is.

This query will not work correctly for the pools with spot price below the old `MinSpotPrice`.

I might have erroneously set `MinSpotPriceV2` in the error. Will fix.

The new query that will work universally for all pools is TBD: https://github.com/osmosis-labs/osmosis/issues/6064

Reply:
I see. I ask because we do lean on `SpotPrice` in volume splitting incentives so there might be an implication here that it wouldn't be possible to retrieve spot price for pools with too small of a spot price

Reply:
Yeah - I thought the conclusion was that this is acceptable for the majority of the pools that we care about for volume splitting. Do you still have any concerns?